### DISCUSSION TITLE
Receive ZIP file from API and write it to the local file system

### DISCUSSION THREAD

**Discussion by feniljariwala82** [nodejs/help#4896] [Q&A] [answered]
Hello guys. I am having a use case of receiving ZIP file from the API server and after receiving I have to write it to the local file system. Can anyone guide me in receiving ZIP file and writing it back to file system? 
Platform: NodeJs v14

**Comment by Trott** [comment]
For questions like this one, see https://github.com/nodejs/node/blob/master/.github/SUPPORT.md#support.

**Comment by feniljariwala82** [comment, marked as answer]
**I tried so hard and got so far in the end it does matter**. LL. I just solved my own problem, feeling proud.

Solution steps:

1. You need to serve your file as a stream in the response "In my case it is a express server which serves ZIP file as a stream"
2. You need to write stream file to the local file system

Express Server Code Example: "Will serve ZIP file when '/file' route hits"

![app](https://user-images.githubusercontent.com/42708686/145766128-67e82ac5-e55f-4148-9bf7-96e70b24384a.png)
![file](https://user-images.githubusercontent.com/42708686/145766156-c147f58a-9fc1-4a60-afd0-3b64e19fb8f9.png)

Receiver Code Example: "Will hit '/file' route and write stream to the file system"

**Make sure you can wrap this code under async function**

![receiver](https://user-images.githubusercontent.com/42708686/145767673-361d94c6-df98-4ca9-8591-e0f506253a3d.png)
